url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveParser_SelectClauseParser.java

```diff
@@ -0,0 +1,7251 @@
+					loop15:
+					while (true) {
+						int alt15=2;
+						int LA15_0 = input.LA(1);
+						if ( (LA15_0==COMMA) ) {
+							alt15=1;
+						}
+
+						switch (alt15) {
+						case 1 :
+							// SelectClauseParser.g:134:56: COMMA identifier
+							{
+							COMMA49=(Token)match(input,COMMA,FOLLOW_COMMA_in_selectItem748);  
+							stream_COMMA.add(COMMA49);
+
+							pushFollow(FOLLOW_identifier_in_selectItem750);
+							identifier50=gHiveParser.identifier();
+							state._fsp--;
+
+							stream_identifier.add(identifier50.getTree());
+							}
+							break;
+
+						default :
+							break loop15;
+						}
+					}
+
+					RPAREN51=(Token)match(input,RPAREN,FOLLOW_RPAREN_in_selectItem754);  
+					stream_RPAREN.add(RPAREN51);
+
+					}
+
+					}
+					break;
+
+			}
+
+			}
+
+			// AST REWRITE
+			// elements: selectExpression, identifier
+			// token labels: 
+			// rule labels: retval
+			// token list labels: 
+			// rule list labels: 
+			// wildcard labels: 
+			retval.tree = root_0;
+			RewriteRuleSubtreeStream stream_retval=new RewriteRuleSubtreeStream(adaptor,"rule retval",retval!=null?retval.getTree():null);
+
+			root_0 = (CommonTree)adaptor.nil();
+			// 135:7: -> ^( TOK_SELEXPR selectExpression ( identifier )* )
+			{
+				// SelectClauseParser.g:135:10: ^( TOK_SELEXPR selectExpression ( identifier )* )
+				{
+				CommonTree root_1 = (CommonTree)adaptor.nil();
+				root_1 = (CommonTree)adaptor.becomeRoot((CommonTree)adaptor.create(TOK_SELEXPR, "TOK_SELEXPR"), root_1);
+				adaptor.addChild(root_1, stream_selectExpression.nextTree());
+				// SelectClauseParser.g:135:41: ( identifier )*
+				while ( stream_identifier.hasNext() ) {
+					adaptor.addChild(root_1, stream_identifier.nextTree());
+				}
+				stream_identifier.reset();
+
+				adaptor.addChild(root_0, root_1);
+				}
+
+			}
+
+
+			retval.tree = root_0;
+
+			}
+
+			retval.stop = input.LT(-1);
+
+			retval.tree = (CommonTree)adaptor.rulePostProcessing(root_0);
+			adaptor.setTokenBoundaries(retval.tree, retval.start, retval.stop);
+
+			 gParent.msgs.pop(); 
+		}
+
+		catch (RecognitionException e) {
+		  throw e;
+		}
+
+		finally {
+			// do for sure before leaving
+		}
+		return retval;
+	}
+	// $ANTLR end "selectItem"
+
+
+	public static class trfmClause_return extends ParserRuleReturnScope {
+		CommonTree tree;
+		@Override
+		public CommonTree getTree() { return tree; }
+	};
+
+
+	// $ANTLR start "trfmClause"
+	// SelectClauseParser.g:138:1: trfmClause : ( KW_MAP selectExpressionList | KW_REDUCE selectExpressionList ) inSerde= rowFormat inRec= recordWriter KW_USING StringLiteral ( KW_AS ( ( LPAREN ( aliasList | columnNameTypeList ) RPAREN ) | ( aliasList | columnNameTypeList ) ) )? outSerde= rowFormat outRec= recordReader -> ^( TOK_TRANSFORM selectExpressionList $inSerde $inRec StringLiteral $outSerde $outRec ( aliasList )? ( columnNameTypeList )? ) ;
+	public final HiveParser_SelectClauseParser.trfmClause_return trfmClause() throws RecognitionException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 199. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.